### ISSUE TITLE
FR: Toasters for custom script blocks.

### ISSUE THREAD

**Issue by irregularisnotrandom** [ukanth/afwall#379]
Hi

Currently we have nessagws for app blocks which is great but ot would be extra nice if AF also shows the blocks done by the custom scripts.

**Comment by CHEF-KOCH** [comment]
I don't see any arguments to add this, because:

* Normal people don't want to be disturbed by million of popups/toasts which shows unnecessary messages, of course that could be solved by disabling this option by default.
* Custom Script users are advance users so they already know what to block and what not, do we really need a feedback of each new established connection that was blocked?
* Instead of blocking xyz the people better uninstall (if possible) the application and use alternatives that he/she trust instead of using any custom script - remember you're smartphone isn't server, blocking xyz "because malware reasons" is just paranoid, you browser i.e. Firefox and other applications already blocking specific 'malware ports' by default (don't know about stock browsers since I never use anything stock - expect for test reasons).
* The mention idea behind is good, but I hear the next guy asking for xyz toasts and it would end up with an on-demand firewall (which truly would be the best for Android) but as written down in the FAQ we don't have enouth manpower for this + the fact Android/Google does not provide any API or want to help on that [I asked several times for giving me examples to introduce this but the answer was always the same -> nothing + labeled as obsolete without anything - thanks Google!].
* Does it makes sense to show something that you can't copy to add e.g. to a whitelist/blacklist? I don't think so.
* We already have an example to log (all?) traffic to a separate .txt file, I suggest something like this because instead of showing the stuff that is maybe blocked it's better to show everything so that the user can decide to add/remove it - similar to the NetworkLog app which shows the entire (?) traffic and AFTER that, the user can decide what todo with it.

**Comment by irregularisnotrandom** [comment]
does the log include the blocks from the custom scripts?

**Comment by CHEF-KOCH** [comment]
No, but you can use an extra command to e.g. log dropped packets into an separate file or directly via <code>var/log/yourlog.log</code>, the path and name must be set into the <code>/etc/syslog.conf</code> file [or that file that you're OS uses (via dmesg or syslogd)].

To log something specific before/after normal iptables are done, you could use <code>--log-prefix together with LOG or TRACE</code> which can be done at anytime in the custom script [but not working both together at the same time!].

But I don't know exactly what you goal should, debugging reasons or just to verify xyz, for the last reasons I better would use an Burp/Wireshark LAN proxy and check the traffic before the app/OS the first time goes online if you try to find out what is generating xyz traffic.
